services/blockchain-indexer/shared/indexer/transactionProcessor/auth/registerMultisignature.js

```diff
@@ -45,16 +47,20 @@ const resolveMultisignatureMemberships = (tx) => {
 			groupAddress: tx.senderAddress,
 		};
 		multisignatureInfoToIndex.push(members);
-
-		// Asynchronously index all the publicKeys
-		updateAccountPublicKey(key);
 	});
 
 	return multisignatureInfoToIndex;
 };
 
 // eslint-disable-next-line no-unused-vars
 const applyTransaction = async (blockHeader, tx, events, dbTrx) => {
+	tx.params.mandatoryKeys.concat(tx.params.optionalKeys).forEach(key => {
+		// Asynchronously index all the publicKeys
+		updateAccountPublicKey(key);
+	});
```

Comment:
To avoid looping through the `optionalKeys` over again.
```suggestion
	// Asynchronously index all the publicKeys
	tx.params.mandatoryKeys.forEach(updateAccountPublicKey);
	tx.params.optionalKeys.forEach(updateAccountPublicKey);
```